integrations/azure_ai_search/src/haystack_integrations/document_stores/azure_ai_search/document_store.py

```diff
@@ -178,6 +189,45 @@ def client(self) -> SearchClient:
 
         return self._client
 
+    def _normalize_metadata_index_fields(
+        self, metadata_fields: Optional[Dict[str, Union[SearchField, type]]]
+    ) -> Dict[str, SearchField]:
```

Comment:
the type hint should be this right?
```suggestion
    ) -> Dict[str, Union[SearchField, SimpleField]]:
```
I see that we convert any `type` values into `SimpleField`

Reply:
Given this https://github.com/deepset-ai/haystack-core-integrations/pull/1893#discussion_r2126467949 it sounds like my suggestion is unnecessary

Reply:
It was a still good point to note 🙏 .